### PR TITLE
Adds Compression to RollFile method

### PR DESCRIPTION
**Pull request by grantwhunt**
As log4net does all the file operations in the background now no extra threading methods were required. We are therefore sipping from the firehose

### CODE REVIEW

File: src/Appender/RollingFileAppender.cs [hunk 1]
```diff
@@ -1283,6 +1287,21 @@ protected void RollFile(string fromFile, string toFile)
...
+					    }
+					    else
+					    {
+					        using ( System.IO.Compression.ZipArchive archive =
+					            System.IO.Compression.ZipFile.Open( toZipFileName, ZipArchiveMode.Create ) )
+					        {
+					            String archivePath = ( $"{System.IO.Path.GetDirectoryName( toFile )}" );
+					            //archive.CreateEntryFromFile( toFile, toZipFileName );
+					            archive.CreateEntryFromFile( toFile, toZipFileName, CompressionLevel.Fastest );
+					            LogLog.Debug( declaringType, $"Archiving {toFile} to {toZipFileName}." );
```
Comment (rabar10) [inline src/Appender/RollingFileAppender.cs]: Minor nit, can you move the LogLog statement to before the archiving action?

Comment (grantwhunt) [reply src/Appender/RollingFileAppender.cs]: Also I was wondering if we should use the CompressionLevel.Optimal instead of Fastest here. Any thoughts?

**Comment by grantwhunt** [comment]
Sorry Rich fat fingered the click I think 😳

**Comment by grantwhunt** [comment]
Im still not convinced this is sufficient as we aren't really taking advantage of the zipping and deleting the old uncompressed log files. Though the previous version of this shim didn't do that either. Or am I missing something here?

**Pull request merged**